Initialize answers inline of the question view.

critocrito opened the pull request:
closes #115 

Answers can be initialized at two places now, inline of the answers
table on the question view and on the edit answer view, the same as
before. I kept the edit page intact since we also use that view to
delete answers.

In order to reduce code duplication I created a new `InitializeAnswer`
components whose purpose it is to manage the logic of initializing
answers. That component is used on the answers table as well as on the
edit answers view. It manages the logic and calls a render
prop (`children`) as a function to render the UI itself.

The table implementation is very limited and doesn't allow for inline
actions (and from what I can tell neither multiple actions on a table
row). I created special support for an `ACTION_ANSWER_INITILIAZE` table
action that is treated special inside the table. If we ever have to
support more inline actions we might want to refactor this approach.

Having multiple actions leads in this case to a rather ugly line break
of the action buttons. Shortening the `Deactivate` label by one
character would solve the problem. Otherwise we need to edit the styles
of the `ButtonOutline` component. I didn't want to do that for this case
since I'm not totally certain of the repercussions to the styling of the
rest of the page.

![Screen Shot 2021-03-10 at 16 51 43](https://user-images.githubusercontent.com/29301569/110657171-85948200-81b8-11eb-9864-c8bf62ee97f8.png)

critocrito commented:
Please also take a look at [this line](https://github.com/bitspossessed/culturestake/pull/155/files#diff-4a73e845fa0ad8e9a3add44743caa1167720ec4a3b9d9a75add651337b8733ccR39). It's the `field` attribute where I defined the association between answers and questions. I'm not sure if I do that correctly here.

ana0 commented on src/server/controllers/answers.js:
```diff
@@ -3,42 +3,57 @@ import baseController from '~/server/controllers';
+    {
+      association: AnswerBelongsToQuestion,
+      destroyCascade: false,
+      fields: ['title', 'chainId'],
```
'chainId' on the question should probably be in fieldsProtected, so that it will be filtered out if this endpoint is ever used without auth

ana0 commented:
I see what you mean about the styling ... it is actually a bit difficult to tell which row the buttons are in when they're split. 

For an alternate (and shorter) word, I'd suggest maybe "disable" but we'd have to change it throughout the ui (there are a number of things that can be "deactivated" currently)

ana0 commented:
Also, I think maybe introduced in this PR since I'm not seeing it on main: on the answer edit page, the copy isn't correctly toggling between "Artwork" and "Property" (for a festival question, where answers are artworks, I'm still seeing property as the label)

critocrito commented:
Hey @ana0, I updated the pull request and I *think* the answer labels should be okay now. This was a great catch, thanks. I also moved the `chainId` to be a protected field.
